src/event_emitter/mod.rs

```diff
@@ -13,32 +13,35 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use std::collections::HashMap;
-use std::sync::{Arc, RwLock};
-
-use crate::api::r0 as api;
-use crate::events::collections::all::{Event, RoomEvent, StateEvent};
-use crate::events::room::{
-    aliases::AliasesEvent,
-    canonical_alias::CanonicalAliasEvent,
-    member::{MemberEvent, MemberEventContent, MembershipState},
-    name::NameEvent,
-};
-use crate::events::EventResult;
-use crate::identifiers::RoomAliasId;
-use crate::session::Session;
+use crate::events::collections::all::RoomEvent;
 use crate::models::Room;
 
-use js_int::{Int, UInt};
-#[cfg(feature = "encryption")]
-use tokio::sync::Mutex;
-
-#[cfg(feature = "encryption")]
-use crate::crypto::{OlmMachine, OneTimeKeys};
-#[cfg(feature = "encryption")]
-use ruma_client_api::r0::keys::{upload_keys::Response as KeysUploadResponse, DeviceKeys};
+// JUST AN IDEA 
+// 
 
+/// This is just a thought I had. Making users impl a trait instead of writing callbacks for events
+/// could give the chance for really good documentation for each event?
+/// It would look something like this
+///
```

Comment:
I can just get rid of this but I thought it might be worth seeing what you think

Reply:
Sounds interesting, though not sure what is more natural for users, adding callbacks or implementing a trait. We should open a separate issue to discuss this.

Though events should be documented in the `ruma-events` crate.